src/wayland-egldisplay.c

```diff
@@ -833,6 +832,12 @@ static bool getServerProtocolsInfo(struct wl_display *nativeDpy,
             }
         }
 
+        /* If we didn't get a name through linux_dmabuf then fall back to wl_drm */
+        if (!protocols->drm_name && protocols->wlDrm) {
+            wl_drm_add_listener(protocols->wlDrm, &drmListener, protocols);
+            wl_display_roundtrip_queue(nativeDpy, queue);
+        }
```

Comment:
Have you actually tested this? Surely at this point the wl_drm events have already been received and discarded by libwayland-client.

Reply:
Yes, this is actually restoring where this `wl_drm_add_listener` has been historically in egl-wayland, before 6355c16.

It works since the earlier roundtrip will bind the the protocols in the registry, but we need a second roundtrip to actually trigger handling of the wl_drm events the compositor sent. This is the problem this change fixes which my previous change had undone. iirc this double roundtrip pattern is present in other projects as well.

Reply:
If zwp_linux_dmabuf_v1 is present, there are three roundtrips and the second one consumes the wl_drm events.

Reply:
Ah I see what you mean now, will fix.